bool direction_forward
int8 speed_percent
